ExchangeSharp/Utility/CryptoUtility.cs

```diff
@@ -1075,37 +1075,39 @@ public static byte[] AesEncryption(byte[] input, byte[] password, byte[] salt)
         /// <param name="password">Password</param>
         /// <param name="salt">Salt</param>
         /// <returns>Decrypted data</returns>
-        public static byte[] AesDecryption(byte[] input, byte[] password, byte[] salt)
+        public static byte[]? AesDecryption(byte[] input, byte[] password, byte[] salt)
         {
             if (input == null || input.Length == 0 || password == null || password.Length == 0 || salt == null || salt.Length == 0)
             {
                 return null;
             }
             MemoryStream decrypted = new MemoryStream();
-            var AES = new RijndaelManaged()
+            using (RijndaelManaged AES = new RijndaelManaged()
             {
                 KeySize = 256,
                 BlockSize = 128,
                 Padding = PaddingMode.PKCS7,
-            };
-            var key = new Rfc2898DeriveBytes(password, salt, 1024);
-            AES.Key = key.GetBytes(AES.KeySize / 8);
-            AES.IV = key.GetBytes(AES.BlockSize / 8);
-            AES.Mode = CipherMode.CBC;
-            MemoryStream encrypted = new MemoryStream(input);
-            byte[] saltMatch = new byte[salt.Length];
-            if (encrypted.Read(saltMatch, 0, saltMatch.Length) != salt.Length || !salt.SequenceEqual(saltMatch))
-            {
-                throw new InvalidOperationException("Invalid salt");
-            }
-            var cs = new CryptoStream(encrypted, AES.CreateDecryptor(), CryptoStreamMode.Read);
-            byte[] buffer = new byte[8192];
-            int count;
-            while ((count = cs.Read(buffer, 0, buffer.Length)) > 0)
-            {
-                decrypted.Write(buffer, 0, count);
+            })
+            {
+                var key = new Rfc2898DeriveBytes(password, salt, 1024);
+                AES.Key = key.GetBytes(AES.KeySize / 8);
+                AES.IV = key.GetBytes(AES.BlockSize / 8);
+                AES.Mode = CipherMode.CBC;
+                MemoryStream encrypted = new MemoryStream(input);
+                byte[] saltMatch = new byte[salt.Length];
+                if (encrypted.Read(saltMatch, 0, saltMatch.Length) != salt.Length || !salt.SequenceEqual(saltMatch))
+                {
+                    throw new InvalidOperationException("Invalid salt");
+                }
+                var cs = new CryptoStream(encrypted, AES.CreateDecryptor(), CryptoStreamMode.Read);
```

Comment:
Maybe here would be good to use `using var` expression.

Reply:
Can you add an example of what you mean? Then I can add a commit to this PR. Alternatively, I can merge this PR and you can put it into your current PR or a new one.

Reply:
I was referring to the same expression in the second example of [this post](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/using-statement#example).
Also, I can add it later :)